src/go/k8s/controllers/cluster.redpanda.com/kgo-config-options.go

```diff
@@ -0,0 +1,228 @@
+
+	// Create Logger
+	kgoLogger := KgoZapLogger{
+		logger: ctrl.Log,
+	}
+	opts = append(opts, kgo.WithLogger(kgoLogger), kgo.WithHooks(hooks))
+
+	if topic.Spec.KafkaAPISpec.SASL != nil {
+		var err error
+		opts, err = configureSASL(topic, opts, log)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// Configure TLS
+	var caCertPool *x509.CertPool
+	if topic.Spec.KafkaAPISpec.TLS == nil {
+		return opts, nil
+	}
+
+	// Root CA
+	if topic.Spec.KafkaAPISpec.TLS.CaCert != nil {
+		var ca corev1.Secret
+		err := cl.Get(ctx, types.NamespacedName{
+			Name:      topic.Spec.KafkaAPISpec.TLS.CaCert.SecretName,
+			Namespace: topic.Namespace,
+		}, &ca)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read ca certificate secret: %w", err)
+		}
+		caCertPool = x509.NewCertPool()
+		isSuccessful := caCertPool.AppendCertsFromPEM(ca.Data[topic.Spec.KafkaAPISpec.TLS.CaCert.SecretKey])
+		if !isSuccessful {
+			log.Info("failed to append ca file to cert pool, is this a valid PEM format?")
+		}
+	}
+
+	// If configured load TLS cert & key - Mutual TLS
+	var certificates []tls.Certificate
+	if topic.Spec.KafkaAPISpec.TLS.Cert != nil && topic.Spec.KafkaAPISpec.TLS.Key != nil {
+		// 1. Read certificates
+		var cert corev1.Secret
+		err := cl.Get(ctx, types.NamespacedName{
+			Name:      topic.Spec.KafkaAPISpec.TLS.Cert.SecretName,
+			Namespace: topic.Namespace,
+		}, &cert)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read certificate secret: %w", err)
+		}
+
+		certData := cert.Data[topic.Spec.KafkaAPISpec.TLS.Cert.SecretKey]
+
+		var key corev1.Secret
+		err = cl.Get(ctx, types.NamespacedName{
+			Name:      topic.Spec.KafkaAPISpec.TLS.Key.SecretName,
+			Namespace: topic.Namespace,
+		}, &key)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read key certificate secret: %w", err)
+		}
+
+		keyData := key.Data[topic.Spec.KafkaAPISpec.TLS.Key.SecretKey]
+
+		// 2. Check if private key needs to be decrypted. Decrypt it if passphrase is given, otherwise return error
+		pemBlock, _ := pem.Decode(keyData)
+		if pemBlock == nil {
+			return nil, fmt.Errorf("no valid private key found") // nolint:goerr113 // this error will not be handled by operator
+		}
+
+		tlsCert, err := tls.X509KeyPair(certData, keyData)
+		if err != nil {
+			return nil, fmt.Errorf("cannot parse pem: %w", err)
+		}
+		certificates = []tls.Certificate{tlsCert}
+	}
+
+	tlsDialer := &tls.Dialer{
+		NetDialer: &net.Dialer{Timeout: 10 * time.Second},
+		Config: &tls.Config{
+			//nolint:gosec // InsecureSkipVerify may be true upon user's responsibility.
+			InsecureSkipVerify: topic.Spec.KafkaAPISpec.TLS.InsecureSkipTLSVerify,
+			Certificates:       certificates,
+			RootCAs:            caCertPool,
+		},
+	}
+
+	return append(opts, kgo.Dialer(tlsDialer.DialContext)), nil
+}
+
+func configureSASL(topic *v1alpha1.Topic, opts []kgo.Opt, log logr.Logger) ([]kgo.Opt, error) {
+	// SASL Plain
+	if topic.Spec.KafkaAPISpec.SASL.Mechanism == config.SASLMechanismPlain {
```

Comment:
This looks like we could have used a switch for the cases here.

Reply:
For me it's subjective and I follow what console did https://github.com/redpanda-data/console/blob/3bfeae1a210b7a7d6b385b95e94a4440610ce543/backend/pkg/kafka/config_helper.go#L65-L158

As I mentioned in https://github.com/RafalKorepta/redpanda/blob/cfd0d673ffaca1f0485b6e73e2682518028e9cdd/src/go/k8s/controllers/cluster.redpanda.com/kgo-config-options.go#L35